src/main/java/com/google/gcs/sdrs/service/impl/JobPoolServiceImpl.java

```diff
@@ -40,12 +73,66 @@ public Integer createJob(PooledJobCreateRequest request) {
 
   @Override
   public Boolean createJobs(Collection<PooledJobCreateRequest> pooledJobCreateRequests) {
+    if (areAllStsJobsValid(pooledJobCreateRequests)) {
+      for (PooledJobCreateRequest pooledJobCreateRequest : pooledJobCreateRequests) {
+        pooledStsJobDao.save(convertToEntity(pooledJobCreateRequest));
+      }
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Method that verifies that for all of the incoming requests 1) the STS Job exists in the cloud
+   * and 2) that the job is not already registered with SDRS.
+   *
+   * <p>If any of the incoming requests fails validation, then the entire batch/transaction of jobs
+   * is rejected
+   *
+   * @param pooledJobCreateRequests
+   * @return
+   */
+  protected boolean areAllStsJobsValid(Collection<PooledJobCreateRequest> pooledJobCreateRequests) {
     for (PooledJobCreateRequest pooledJobCreateRequest : pooledJobCreateRequests) {
-      pooledStsJobDao.save(convertToEntity(pooledJobCreateRequest));
+      if (!doesJobExist(pooledJobCreateRequest) || isJobAlreadyRegistered(pooledJobCreateRequest)) {
+        return false;
+      }
     }
     return true;
   }
-  
+
+  protected boolean doesJobExist(PooledJobCreateRequest pooledJobCreateRequest) {
+    try {
+      TransferJob transferJob =
+          StsUtil.getExistingJob(
+              client, pooledJobCreateRequest.getProjectId(), pooledJobCreateRequest.getName());
+      if (transferJob == null || transferJob.isEmpty()) {
+        return false;
+      }
+      return true;
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
consider using RetentionUtil.convertStackTrace. https://github.com/GoogleCloudPlatform/storage-sdrs/blob/develop/src/main/java/com/google/gcs/sdrs/service/worker/rule/impl/StsRuleExecutor.java#L201-L206 as e.printStackTrace() creates multiple log entries in StackDriver